Filer.py

```diff
@@ -312,6 +329,70 @@ def make_tree(rel, pathname, clean_expired=True):
     return tree
 
 
+def read_users(htpasswd_file):
+    users = []
+    if path.exists(htpasswd_file):
+        with open(htpasswd_file, "r", encoding="utf-8") as htpasswd_in:
+            for line in htpasswd_in.readlines():
+                splitted_line = line.split(":", 1)
+                if len(splitted_line) == 2:
```

Comment:
This one is hard to understand and requires some knowledge not at hand at this moment.